packages/cursorless-vscode-e2e/src/suite/performance.vscode.test.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  asyncSafety,
+  type ActionDescriptor,
+  type ScopeType,
+  type SimpleScopeTypeType,
+} from "@cursorless/common";
+import { openNewEditor, runCursorlessCommand } from "@cursorless/vscode-common";
+import assert from "assert";
+import * as vscode from "vscode";
+import { endToEndTestSetup } from "../endToEndTestSetup";
+
+const testData = generateTestData();
+const numLines = testData.split("\n").length;
+
+suite(`Performance: ${numLines} lines JSON`, async function () {
+  endToEndTestSetup(this);
+
+  let previousTitle = "";
+
+  this.beforeEach(function () {
+    const title = this.currentTest!.title;
+    if (title !== previousTitle) {
+      console.log(`    ${title}`);
+      previousTitle = title;
+    }
+  });
```

Comment:
what's this for? doesn't mocha print the test name? I guess for the slow test you want to know what it's doing while you wait?

Reply:
Mocha prints the name after the test is done. And if the test fails I want the test title and then each run timer. Look at the example in the description.